### ISSUE TITLE
I have trying to compile pdf in Indic language.

### ISSUE THREAD

**Issue by IDontKnow2Code** [aligrudi/neatroff_make#16]
Hello,
I have been trying to compile documents in Tamil (one of the Indian Languages). But neatroff outputs a blank pdf.
Font I used [MuktaMalar](https://fonts.google.com/specimen/Mukta+Malar?subset=tamil#about) 
Here is the file [Tamil.ms](http://0x0.st/-piB.ms)
I m new to troff, so please forgive me if it is something simple. 
Thank you.

**Comment by aligrudi** [comment]
Hello,

IDontKnow2Code ***@***.***> wrote:
> I have been trying to compile documents in Tamil (one of the Indian Languages). But neatroff outputs a blank pdf.
> Font I used [MuktaMalar](https://fonts.google.com/specimen/Mukta+Malar?subset=tamil#about) 
> Here is the file [Tamil.ms](http://0x0.st/-piB.ms)
> I m new to troff, so please forgive me if it is something simple. 

First of all, note that macros like TL, AU, SH, and PP change
the current font.  Therefore, you should change the font after the
invocation of each of these macros, redefine these macros, or
use the following line to redefine R and B fonts:

  .fp - R MuktaMalar-Regular
  .fp - B MuktaMalar-Bold

About the output, there seems to be a problem with Ghostscript.
Try building Tamil.PDF in neatroff_make/demo directory (with
uppercase letters in the extension); this uses the PDF
postprocessor of Neatroff (produces the PDF output directly and
does not use Ghostscript).  Does this work?

	Ali

**Comment by IDontKnow2Code** [comment]
I removed TL,AU,SH,LP and previous font definition and redefined fonts as you said and compiled it using demo make with .PDF, I have a output but the words are overlapping

**Comment by aligrudi** [comment]
IDontKnow2Code ***@***.***> wrote:
> I removed TL,AU,SH,LP and previous font definition and redefined
> fonts as you said and compiled it using demo make with .PDF, I have a
> output but the words are overlapping

The problem with the font is that it does not include PostScript
font name.  I changed Neatpost to use its file name, if font name
is missing.  After this change, the PostScript backend of Neatpost
should also work.

About the overlapping words, can you include a very small
example plus the way it should actually look.

	Ali

**Comment by IDontKnow2Code** [comment]
By example you mean how the letters should actually look?

**Comment by IDontKnow2Code** [comment]
This is how it should look 
![tamilC](https://user-images.githubusercontent.com/77267330/123433038-a908fe80-d5e8-11eb-8c19-7648db7e313b.png)
But this is how it outputs 
![tamilInc](https://user-images.githubusercontent.com/77267330/123433098-b7571a80-d5e8-11eb-8f61-a5cff22607af.png)
Title in both images are same please check that to know the differences.

**Comment by aligrudi** [comment]
IDontKnow2Code ***@***.***> wrote:
> This is how it should look 
> ![tamilC](https://user-images.githubusercontent.com/77267330/123433038-a908fe80-d5e8-11eb-8c19-7648db7e313b.png)
> But this is how it outputs 
> ![tamilInc](https://user-images.githubusercontent.com/77267330/123433098-b7571a80-d5e8-11eb-8f61-a5cff22607af.png)
> Title in both images are same please check that to know the differences.

Thanks.

Actually rendering Tamil text is more complex than what I thought:
  https://docs.microsoft.com/en-us/typography/script-development/tamil

OpenType GPOS and GSUB font features can be enabled with .ff request (Do
you see any improvements?):
  .ff R +nukt +akhn +rphf +rkrf +pref +blwf +half +pstf +cjct +init +pres +abvs +blws +ptts +haln +psts +dist +abvm +blwm

However, that does not seem to be enough for Tamil.

	Ali

**Comment by IDontKnow2Code** [comment]
Yeah I see some improvements.
![tamilSC](https://user-images.githubusercontent.com/77267330/123503990-12821f00-d674-11eb-9b9a-3f5047453e48.png)
But like you said its not complete, should I try using any other type of fonts like .pfb or .afm?

**Comment by aligrudi** [comment]
IDontKnow2Code ***@***.***> wrote:
> Yeah I see some improvements.
> ![tamilSC](https://user-images.githubusercontent.com/77267330/123503990-12821f00-d674-11eb-9b9a-3f5047453e48.png)
> But like you said its not complete, should I try using any other type of fonts like .pfb or .afm?

I have not read the details yet.  However, it seems rather complex,
especially for someone unfamiliar with this script; see the shaping
engine section of the following page:

  https://docs.microsoft.com/en-us/typography/script-development/tamil

Can you summarize what should be done besides applying the standard
OpenType features?

Thanks,
Ali

**Comment by IDontKnow2Code** [comment]
I think `matra` is the problem here.
They are responsible for (as far as what I understand) க,கா,கி,கீ,கு,கூ,கெ,கே,கை,கொ,கோ,கௌ.(these are combination of all vowels and a single consonant)
The problem here(in pdf we created) is க,கா,கி,கீ,கு,கூ seems to be forming correct, but others கெ,கே,கை,கொ,கோ,கௌ are not forming as it should be.
